db/cache_db.go

```diff
@@ -37,63 +41,77 @@ func (db *CacheDB) Get(key []byte) []byte {
 	return db.parent.Get(key)
 }
 
+// Set sets the value for the given key in the cache. It does not write the value to the
+// underlying database. Set does not incur a performance penalty compared to SetNoLock.
```

Comment:
What's all this about performance penalty? These functions lock a mutex while XxxNoLock does not